opencog/nlp/aiml/aiml.scm

```diff
@@ -468,8 +467,19 @@
 	(let ((response (word-list-flatten (do-while-same SENT 5))))
 
 		; The robots response is the current "that".
+		; Store up to two previous inputs and outputs
+		; XXX TODO: Would be better to log and retrieve the chat
+		;           history using AtTimeLink and the time server
 		(if (valid-response? response)
-			(do-aiml-set (Concept "that") response))
+			(let ((that (do-aiml-get (Concept "that"))))
```

Comment:
never mind!  I see that `do-aiml-get` already does this, automatically.   Kind of weird, though :-(